### DISCUSSION TITLE
Ability to hide bookmarks page

### DISCUSSION THREAD

**Discussion by discoverlance-com** [STAFE-GROUP-AB/filament-bookmarks-menu#2] [Ideas]
I am currently using the package and it's very good but I would really love it if it's configurable in the `config/filament-bookmarks-menu.php` file to exclude the page from the navigation in the filament menu.

I basically want the bookmarks to only be accessible from the bookmarks button at the top. Having the users see the bookmarks as a page is not really ideal in my case especially as the form can be a bit intimidating to a user as to what it means like target and also the form allowing them to select who the bookmark belongs to (which means I can create a bookmark and assign it to someone else) and even if we tell them not to use the form and to instead use the bookmark button to add a new one, it does not make sense to have the bookmark page appear in the menu. It will be good if we can just disable the page on the filament menu from the config or even disable the create button (New bookmark) on the bookmarks page.

We can move the management of the bookmark to the bookmark dropdown on the filament topbar. So we can have edit and delete buttons just next to each bookmark to easily update or delete a bookmark without going to the bookmarks page.

**Comment by andreaskviby** [comment]
Hey
Have you tried to use filament shield? With that you can show / hide resources and actions the way you want. I will try to figure out how to solve this for those who don’t use it.

**Reply by discoverlance-com** [reply]
I see, yh I am using filament shield. 
But when I try to remove the view or view any permission from bookmarks it still shows in the menu and I am able to access the bookmarks. So I guess it's not working.

Also as a user, I think I should only see the bookmarks that I created and when I am creating on the bookmarks page, I think there should be no select for the user but it should be auto assigned to me. As at now, if a user goes to the bookmarks page, they will see bookmarks belonging to several users and they might have to search in other to find theirs which is not a good experience I think.

**Reply by andreaskviby** [reply]
I guess you are not impersonating a user or you can add a scope. But I think your wish is good so I will make that in the new version.

**Reply by discoverlance-com** [reply]
> I guess you are not impersonating a user or you can add a scope. But I think your wish is good so I will make that in the new version.

How do I add a scope then. I am not impersonating a user and I am using the current user to try to disable at least the button to add a new bookmark for now. How do I disable that  or use scope to resolve this. I am using shield by the way...

**Comment by andreaskviby** [comment]
Hey,
The version 1.0.3 is now released and in the config you can now add this.
'show_resource_in_navigation' => false,
Then the resource will not be visible

**Reply by discoverlance-com** [reply]
Ok I have seen that. I just upgraded my package. Thanks very much. 

Can I make this contribution, thus updating the README to include this new feature details?

I think now if we can be able to delete the bookmark with a delete icon just next to the dropdown in the topbar, that could help so we in essence can manage our bookmarks just from the dropdown without requiring the actual bookmark page.

**Reply by andreaskviby** [reply]
Will do that also

**Reply by andreaskviby** [reply]
Hey
Now the delete button is in place, just update and you will see. The readme is also updated as it should

**Reply by discoverlance-com** [reply]
Ok thanks very much for this update, will check it out.

**Reply by andreaskviby** [reply]
Did it work the way you wanted?

**Reply by discoverlance-com** [reply]
> Did it work the way you wanted?

Hi, I can see the delete button alright but whilst trying to delete, I get the error `Class "App\Models\FilamentBookmarksMenu" not found`

**Reply by andreaskviby** [reply]
That’s weird because the model do exist in your application

**Reply by discoverlance-com** [reply]
I think the publish command, `php artisan vendor:publish --tag="filament-bookmarks-menu-config"` only publishes the config and the migration (`php artisan vendor:publish --tag="filament-bookmarks-menu-migrations"`) right?

So in `src/Http/Livewire/BookmarksMenu.php` line 45, it should rather use the `\STAFEGROUPAB\FilamentBookmarksMenu\Models\BookmarksMenu` model as the model is not published to my `App\Models` namespace but it's in the package.

I tried directly changing that model to `\STAFEGROUPAB\FilamentBookmarksMenu\Models\BookmarksMenu` in the `vendor/stafe-group-ab/filament-bookmarks-menu/src/Http/Livewire/BookmarksMenu.php` and the deletion worked.

**Reply by andreaskviby** [reply]
Ok will update

**Reply by discoverlance-com** [reply]
> Ok will update

Thanks. I just checked the update and I am able to successfully delete the bookmark.

**Reply by andreaskviby** [reply]
Great
